src/unix/lwt_unix_unix.h

```diff
@@ -3009,38 +3014,200 @@ CAMLprim value lwt_unix_bind_job(value fd, value address)
+       runtime representation to normal C, machine representation, passes them
+       to `read(2)`, converts the result back to OCaml, and returns.
+
+    2. In case the underlying C call is blocking, as `getcwd(3)` is, Lwt
+       "converts" it to a non-blocking call by running it inside a worker
+       thread. The rest of this comment is concerned with such blocking calls.
+
+    For background on writing C stubs in OCaml, see
+
+      http://caml.inria.fr/pub/docs/manual-ocaml/intfc.html
+
+
+    Each Lwt stub for a blocking C call defines a *job* for the Lwt worker
+    thread pool. The actual thread pool is implemented in `lwt_unix_stubs.c` and
+    is beyond the scope of this comment. It is not necessary to understand it to
+    implement Lwt jobs (indeed, the author currently doesn't remember exactly
+    how it works!).
+
+    A job is defined by providing three functions. Let's say we want to call our
+    new kind of job "`FOO`":
+
+    - `lwt_unix_FOO_job` allocates the job `struct` (see next paragraph).
+    - `worker_FOO` is later called in a worker thread to actually run the job.
+    - `result_FOO` is, even later, called in the main thread to return the
+      result of the job to OCaml, and deallocate the job.
+
+    It is also necessary to define a `struct` with name `struct job_FOO`. This
+    is the representation of the job that will be manipulated by the thread
+    pool. It has several purposes:
+
+    - Store the pointers to `worker_FOO` and `result_FOO`, so the thread pool is
+      able to run them.
+    - Store the C call arguments, or references to them, so they can be accessed
+      by `worker_FOO` when it runs in the worker thread.
+    - Store the C call results, so they can be accessed by `result_FOO` in the
+      main thread.
+    - Be something that can be placed in queues, or manipulated otherwise.
+
+    So, there are three functions and a `struct` to define. It is important that
+    their names fit the pattern given above, because it is expected by Lwt job
+    helper macros.
+
+    It is also possible to define additional helper functions and/or types, as
+    needed.
+
+
+    Many stubs are defined on Unix-like systems, but not Windows, and vice
+    versa. However, the OCaml code lacks conditional compilation, and expects
+    all the C symbols to be available on all platforms during linking – it just
+    doesn't call the ones that aren't really defined.
+
+    The `LWT_NOT_AVAILABLEx` macros are used to define dummy symbols. `x` is the
+    number of arguments the stub takes. For example, the `getcwd` job is
+    currently not defined on Windows. For this reason, `lwt_unix_windows.h` has
+    `LWT_NOT_AVAILABLE1(unix_getcwd_job)`. The `lwt_` prefix is left off.
+
+    A typical way to discover that you need to use this kind of macro is to
+    submit a PR to the Lwt repo, and wait for the CI build bots to tell you. We
+    have pretty thorough CI coverage of the systems Lwt supports.
```

Comment:
Ideally, yes, but in practice I would forget those macros, even after setting up the CI and watching it fail on my code and others' code in the past. But that's a comment on whether the documentation while make the CI-driven development obsolete. I agree the documentation itself should be changed :)